codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanDecoder.java

```diff
@@ -33,214 +33,4685 @@
 
 import io.netty.buffer.ByteBuf;
 import io.netty.util.AsciiString;
-import io.netty.util.ByteProcessor;
-import io.netty.util.internal.ObjectUtil;
 import io.netty.util.internal.ThrowableUtil;
 
 import static io.netty.handler.codec.http2.Http2Error.COMPRESSION_ERROR;
 
 final class HpackHuffmanDecoder {
 
-    private static final Http2Exception EOS_DECODED = ThrowableUtil.unknownStackTrace(
-            Http2Exception.newStatic(COMPRESSION_ERROR, "HPACK - EOS Decoded",
-                    Http2Exception.ShutdownHint.HARD_SHUTDOWN), HpackHuffmanDecoder.class, "decode(..)");
-    private static final Http2Exception INVALID_PADDING = ThrowableUtil.unknownStackTrace(
-            Http2Exception.newStatic(COMPRESSION_ERROR, "HPACK - Invalid Padding",
-                    Http2Exception.ShutdownHint.HARD_SHUTDOWN), HpackHuffmanDecoder.class, "decode(..)");
+    /* Scroll to the bottom! */
 
-    private static final Node ROOT = buildTree(HpackUtil.HUFFMAN_CODES, HpackUtil.HUFFMAN_CODE_LENGTHS);
+    private static final byte HUFFMAN_COMPLETE = 1;
+    private static final byte HUFFMAN_EMIT_SYMBOL = 1 << 1;
+    private static final byte HUFFMAN_FAIL = 1 << 2;
 
-    private final DecoderProcessor processor;
```

Comment:
now as this only contains static methods you should also add a private constructor.